marimo-tutorials/artificial-intelligence/deep-learning/activation-functions.py

```diff
@@ -0,0 +1,382 @@
+# /// script
+# requires-python = ">=3.12"
+# dependencies = [
+#     "marimo",
+#     "anywidget==0.9.13",
+#     "traitlets==5.14.3",
+#     "numpy==2.1.2",
+#     "matplotlib==3.9.2",
+# ]
+# ///
+"""
+This is the starting point for your notebook.
+"""
+
+import marimo
+
+__generated_with = "0.9.14"
+app = marimo.App(app_title="Activation-Functions")
+
+
+@app.cell(hide_code=True)
+def __(header_widget):
+    header_widget
+    return
+
+
+@app.cell(hide_code=True)
+def __(activation_functions, mo):
+    mo.md(
+        f"""
+        ## Activation Functions
+            [Activation functions](https://en.wikipedia.org/wiki/Activation_function) are crucial components in neural networks that introduce 
+            non-linearity into the network's learning process. They determine whether a 
+            neuron should be fired/activated or not by calculating the weighted sum and adding bias.
+            Each function has its unique properties and use cases in different scenarios.
+            Select an activation function: {activation_functions}
+        """
+    )
+    return
+
+
+@app.cell(hide_code=True)
+def __(activation_functions, mo):
+    import numpy as np
+    import matplotlib.pyplot as plt
+
+    @mo.cache(pin_modules=True)  # if module versions change later on
+    def plot_activation(func_name):
+        x = np.linspace(-10, 10, 1000)
+        y = np.zeros_like(x)  # Initialize y array
+        formula = ""
+        
+        if func_name == "sigmoid":
+            y = 1 / (1 + np.exp(-x))
+            formula = r"$$\sigma(x) = \frac{1}{1 + e^{-x}}$$"
+        elif func_name == "tanh":
+            y = np.tanh(x)
+            formula = r"$$tanh(x) = \frac{e^x - e^{-x}}{e^x + e^{-x}}$$"
+        elif func_name == "relu":
+            y = np.maximum(0, x)
+            formula = r"$$ReLU(x) = max(0, x)$$"
+        elif func_name == "elu":
+            alpha = 1.0  # Default alpha value for ELU
+            y = np.where(x > 0, x, alpha * (np.exp(x) - 1))
```

Comment:
For activation functions with hyper-parameters, we can add another slider to let reader control values

Reply:
_Fair_.